src/Bicep.Core/Syntax/SyntaxRewriteVisitor.cs

```diff
@@ -251,14 +252,15 @@ protected virtual SyntaxBase ReplaceResourceDeclarationSyntax(ResourceDeclaratio
             hasChanges |= TryRewrite(syntax.Type, out var type);
             hasChanges |= TryRewriteStrict(syntax.ExistingKeyword, out var existingKeyword);
             hasChanges |= TryRewrite(syntax.Assignment, out var assignment);
+            hasChanges |= TryRewrite(syntax.Newlines, out var newlines);
             hasChanges |= TryRewrite(syntax.Value, out var value);
 
             if (!hasChanges)
             {
                 return syntax;
             }
 
-            return new ResourceDeclarationSyntax(leadingNodes, keyword, name, type, existingKeyword, assignment, value);
+            return new ResourceDeclarationSyntax(leadingNodes, keyword, name, type, existingKeyword, assignment, newlines.Cast<Token>().ToImmutableArray(), value);
```

Comment:
>Cast [](http://example.com/codeflow?start=122&length=4)

Are there any syntax rewrite scenarios that could cause the cast to fail?

Reply:
No, the rewrite method should not change the newline token type. If it does then it's a bug.